src/test/librbd/object_map/test_mock_DiffRequest.cc

```diff
@@ -234,7 +234,7 @@ TEST_F(TestMockObjectMapDiffRequest, IntermediateDelta) {
   BitVector<2> expected_diff_state;
   expected_diff_state.resize(object_count);
   expected_diff_state[1] = DIFF_STATE_DATA;
-  expected_diff_state[2] = DIFF_STATE_DATA_UPDATED;
+  expected_diff_state[2] = DIFF_STATE_DATA;
```

Comment:
Why did you change this? The initial state is `OBJECT_EXISTS_CLEAN` and the final state is `OBJECT_EXISTS`, so the object is updated and the diff state should be `DIFF_STATE_DATA_UPDATED`. With this change the test expectly fails.

In my comment I meant that e.g. in this test if you changed the `object_map_2[2] = OBJECT_EXISTS` to `object_map_2[2] = OBJECT_EXISTS_CLEAN` then yes, the diff state would be `DIFF_STATE_DATA` (not updated). But instead of modifying this `OBJECT_EXISTS_CLEAN -> OBJECT_EXISTS` case (which is needed to be tested too) I would extend the object map (increase `object_count`) and include this additional `OBJECT_EXISTS -> OBJECT_EXISTS_CLEAN` scenario.

Reply:
Sorry, its wrong patch, I pushed my old branch.